packages/fxa-auth-server/lib/senders/renderer/index.ts

```diff
@@ -0,0 +1,186 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { DOMLocalization, Localization } from '@fluent/dom';
+import { RendererBindings, TemplateContext } from './bindings';
+import Localizer, { FtlIdMsg } from '../../l10n';
+
+const RTL_LOCALES = [
+  'ar',
+  'ckb',
+  'dv',
+  'he',
+  'ks',
+  'ps',
+  'fa',
+  'syr',
+  'ur',
+  'ug',
+];
+interface GlobalTemplateValues {
+  subject: FtlIdMsg;
+  action?: FtlIdMsg;
+}
+
+class Renderer extends Localizer {
+  protected readonly bindings: RendererBindings;
+
+  constructor(bindings: RendererBindings) {
+    super(bindings);
+    this.bindings = bindings;
+  }
+
+  private async localizeAndRender(
+    l10n: DOMLocalization | Localization,
+    string: FtlIdMsg,
+    context: TemplateContext
+  ) {
+    const localizedString =
+      (await l10n.formatValue(string.id, context)) || string.message;
+    return localizedString.includes('<%')
+      ? this.bindings.renderEjs(localizedString, context)
+      : localizedString;
+  }
+
+  /**
+   * Renders and localizes an MJML/EJS email.
+   * @param context Contains either values sent through mailer.send or mock values from Storybook
+   * @returns html HTML transformed from MJML that is rendered through EJS and localized
+   * @returns text Plaintext rendered through EJS and localized
+   * @returns subject Localized subject, for mailer use
+   */
+  async renderEmail(context: TemplateContext) {
+    const { acceptLanguage, template, layout } = context;
+    const { l10n, selectedLocale } = await super.setupLocalizer(
+      acceptLanguage,
+      true
+    );
+    // emails are sent with a `templateValues` object (though not in Storybook) but we spread
+    // them here to make them more top-level accessible
+    context = { ...context, ...context.templateValues };
+    // cssPath is relative to where rendering occurs
+    context.cssPath = this.bindings.opts.templates.cssPath;
+
+    if (template !== '_storybook') {
+      /*
+       * TODO: finish pulling subject & actions out of mailer and into template files, then
```

Comment:
I really made this section so much messier, but it's going to be nice after #11960